File: backend/src/main/java/com/healthcare/radiology_manager/service/EquipmentService.java
package com.healthcare.radiology_manager.service;

import com.healthcare.radiology_manager.dto.EquipmentCreateRequest;
import com.healthcare.radiology_manager.dto.EquipmentResponse;


public interface EquipmentService {

    EquipmentResponse createEquipment(EquipmentCreateRequest request);
}

File: radiology-manager-fe/src/app/services/equipment.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Equipment, EquipmentCreateRequest } from '../shared/models/equipment.model';

@Injectable({ providedIn: 'root' })
export class EquipmentService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiBaseUrl}/equipment`;

  createEquipment(request: EquipmentCreateRequest): Observable<Equipment> {
    return this.http.post<Equipment>(this.baseUrl, request);
  }
}
